src/ray/raylet/task_dependency_manager.cc

```diff
@@ -169,7 +214,62 @@ bool TaskDependencyManager::SubscribeDependencies(
   return (task_entry.num_missing_dependencies == 0);
 }
 
-bool TaskDependencyManager::UnsubscribeDependencies(const TaskID &task_id) {
+void TaskDependencyManager::RemoveTaskDependency(const TaskID &task_id,
+                                                 const ObjectID &object_id) {
+  // Remove the task from the list of tasks that are dependent on this
+  // object.
+  // Get the ID of the task that creates the dependency.
+  TaskID creating_task_id = ComputeTaskId(object_id);
+  auto creating_task_entry = required_tasks_.find(creating_task_id);
+  auto &dependent_tasks = creating_task_entry->second[object_id];
+  size_t erased = dependent_tasks.erase(task_id);
+  RAY_CHECK(erased > 0);
+  // If the unsubscribed task was the only task dependent on the object, then
+  // erase the object entry.
+  if (dependent_tasks.empty()) {
+    creating_task_entry->second.erase(object_id);
+    // Remove the task that creates this object if there are no more object
+    // dependencies created by the task.
+    if (creating_task_entry->second.empty()) {
+      required_tasks_.erase(creating_task_entry);
+    }
+  }
+}
+
+bool TaskDependencyManager::UnsubscribeGetDependencies(const TaskID &task_id) {
+  // Remove the task from the table of subscribed tasks.
+  auto it = task_dependencies_.find(task_id);
+  if (it == task_dependencies_.end()) {
+    return false;
+  }
+
+  // Update the task to reflect that it is no longer blocked on any objects.
+  // It may still have active ray.wait calls, so we cannot remove it
+  // immediately.
+  TaskDependencies &task_entry = it->second;
+  const auto get_dependencies = std::move(task_entry.get_dependencies);
+  task_entry.get_dependencies.clear();
```

Comment:
This is a C++ question, but is it ok to use this object after calling `std::move` on it?

Reply:
Yes, according to the reference:
```
Unless otherwise specified, all standard library objects that have been moved from are placed in a valid but unspecified state. That is, only the functions without preconditions, such as the assignment operator, can be safely used on the object after it was moved from: 
```

`clear()` has no preconditions.